integration-test-common/src/test/java/org/apache/gravitino/integration/test/util/DownloaderUtils.java

```
@@ -61,8 +79,106 @@ public static void downloadFile(String fileUrl, String... destinationDirectories
     }
   }
 
-  private static String getFileName(URL url) {
-    String[] pathSegments = url.getPath().split("/");
+  /**
+   * Check and clean driver version conflicts in directories
+   *
+   * @param targetUrl Target driver URL
+   * @param directories Directories to check
+   * @throws IOException If file operations fail
+   */
+  private static void checkAndCleanDriverConflicts(String targetUrl, String... directories)
+      throws IOException {
+    String expectedFileName = getFileName(targetUrl);
+    String expectedVersion = extractVersion(expectedFileName);
+    String driverType = getDriverType(expectedFileName);
+
+    if (expectedVersion == null || driverType == null) {
+      LOG.warn("Unable to extract driver version or type from URL: {}", targetUrl);
+      return;
```

when will this happen? should we throw an exception instead of a direct return?